h/static/styles/partials-v2/_search.scss

```diff
@@ -367,6 +375,10 @@ $stats-icon-column-width: 20px;
 
   .search-result-bucket__annotation-cards {
     width: 100%;
+
+    // Set the preferred height of the annotation card list based on its
+    // contents.
+    flex-basis: auto;
```

Comment:
On smaller screens the bucket sidebar and card list are arranged vertically and `flex-basis` controls the height, which we want to be based on the card's contents.